src/components/tl/mlx5/mcast/tl_mlx5_mcast_rcache.c

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5_mcast_rcache.h"
+
+static ucs_status_t ucc_tl_mlx5_mcast_coll_reg_mr(ucc_tl_mlx5_mcast_coll_context_t *ctx,
```

Comment:
why return ucs_status_t?

Reply:
this is because of ucc_tl_mlx5_mcast_rcache_mem_reg_cb callback. It must return UCS statuses, otherwise it does not compile.